dvc/parsing/interpolate.py

```diff
@@ -203,6 +203,16 @@ def validate_value(value, key):
     if not_primitive and not_foreach:
         if isinstance(value, dict) and key == "cmd":
             return True
+        if isinstance(value, list) and key in [
+            "deps",
+            "outs",
+            "params",
+            "metrics",
+            "plots",
+        ]:
+            for e in value:
+                validate_value(e, key)
```

Comment:
The rationale for why I did it this way was strictly simplicity, and because it follows what you already did for dict unpacking.

I'm open to re-validating but would need to take another look to figure out how to do it (feel free to try it). The concern about error handling was that the errors won't match what the users sees in `dvc.yaml` since this would be a sort of virtual copy with interpolated values expanded, right?

Reply:
Makes sense. To clarify, I don't consider it a blocker.

Reply:
> i.e. frozen must be bool

That way, you have to reimplement all of the schema validation. I don't really like what we do for `cmd` already. It gets harder too, eg: `outs` could be a list of dict with specific values.